crates/noirc_evaluator/src/ssa_refactor/opt/unrolling.rs

```diff
@@ -241,11 +243,11 @@ fn unroll_loop_header<'a>(
                 // unroll_into block from now on.
                 context.insert_block = unroll_into;
 
-                loop_.blocks.contains(&context.source_block).then_some(context)
+                Ok(loop_.blocks.contains(&context.source_block).then_some(context))
             } else {
                 // If this case is reached the loop either uses non-constant indices or we need
                 // another pass, such as mem2reg to resolve them to constants.
-                None
+                Err(())
```

Comment:
This returning a `None` value instead of `Err` was signalling the loop was finished unrolling (successfully) rather than finished with an error.

This lead to the loop still being present in the program and kept being picked up again by find_all_loops. Signalling an error instead adds it to the list of loops that failed to unroll and prevents it from being unrolled again.